lib/phoenix/tracker/state.ex

```diff
@@ -252,14 +266,21 @@ defmodule Phoenix.Tracker.State do
   end
 
   @spec observe_removes(t, t, [value]) :: {clouds, delta, leaves :: [value]}
-  defp observe_removes(%State{pids: pids, values: values} = local, remote, remote_map) do
+  defp observe_removes(%State{pids: pids, values: values, delta: delta} = local, remote, remote_map) do
     unioned_clouds = union_clouds(local, remote)
-    init = {unioned_clouds, local.delta, []}
-
-    foldl(local.values, init, fn {{topic, pid, key}, _, tag} = el, {clouds, delta, leaves} ->
-      if in?(remote, tag) and not Map.has_key?(remote_map, tag) do
-        1 = :ets.select_delete(values, [{el, [], [true]}])
-        1 = :ets.select_delete(pids, [{{pid, topic, key}, [], [true]}])
+    %State{context: remote_context, clouds: remote_clouds, replica: replica} = remote
+    init = {unioned_clouds, delta, []}
+    # fn {key, meta, {^replica, clock}} -> {key, meta, {replica, clock}} end
```

Comment:
In the previous code we didn't match on the replica. Are you matching it here because `in?` is going to filter on it anyway?

Reply:
That's what provides the improvement and leverages the fact that our CRDT is special in that we'll only get downs from that particular replica when processing the delta from that replica.

Reply:
Beautiful!